packages/algorand-js/src/interpreter/opcode-list.ts

```diff
@@ -1602,3 +1602,160 @@ export class Addr extends Op {
     stack.push(addr.publicKey);
   }
 }
+
+// get balance for the requested account specified
+// by Txn.Accounts[A] in microalgos. A is specified as an account
+// index in the Accounts field of the ApplicationCall transaction,
+// zero index means the sender
+export class Balance extends Op {
+  readonly interpreter: Interpreter;
+
+  /**
+   * @param args words list extracted from line: {}
+   * @param line line number in TEAL file
+   * @param interpreter Interpreter Object
+   */
+  constructor (args: string[], line: number, interpreter: Interpreter) {
+    super();
+    this.interpreter = interpreter;
+
+    assertLen(args.length, 0, line);
+  };
+
+  execute (stack: TEALStack): void {
+    this.assertMinStackLen(stack, 1);
+    let last = this.assertBigInt(stack.pop());
+    let bal = 0;
+
+    if (last === BigInt("0")) {
+      const val = this.interpreter.accounts.get(convertToString(this.interpreter.tx.snd));
+      if (val === undefined) {
+        throw new TealError(ERRORS.TEAL.ACCOUNT_DOES_NOT_EXIST);
+      }
+      bal = val.amount;
+    } else {
+      this.checkIndexBound(Number(--last), this.interpreter.tx.apat);
+
+      const buf = this.interpreter.tx.apat[Number(last)];
+      const res = this.interpreter.accounts.get(convertToString(buf));
+
+      if (res === undefined) {
+        throw new TealError(ERRORS.TEAL.ACCOUNT_DOES_NOT_EXIST);
+      }
+
+      bal = res.amount;
+    }
+
+    stack.push(BigInt(bal));
+  }
+}
+
+// get Asset Holding Info for given account
+// read from account specified by Txn.Accounts[A]
+// and asset B holding field X (imm arg) => {0 or 1 (top), value}
+// params: account index, asset id. Return: did_exist flag (1 if exist and 0 otherwise), value.
```

Comment:
```suggestion
// For ASA A, Account B (txn.accounts[B]) pushes to the 
// stack {...stack, 0} if account has no A holding, 
// otherwise {...stack, holding, 1}
```